src/objects/selection.ts

```diff
@@ -49,8 +55,28 @@ export function selectItem(item: GraphItemSelection | null): void {
   emitter.emit('itemSelected', item)
 }
 
-export function isSelected(item: GraphItemSelection | SelectableItem | null): boolean {
-  return item?.id === selected?.id
+export function isSelected(item: GraphItemSelection): boolean {
+  if (selected === null) {
+    return false
+  }
+
+  const { kind } = item
+
+  switch (kind) {
+    case 'task-run':
+      return isNodeSelection(selected) && selected.id === item.id
+    case 'flow-run':
+      return isNodeSelection(selected) && selected.id === item.id
+    case 'artifact':
+      return isArtifactSelection(selected) && selected.id === item.id
+    case 'artifacts':
+      return isArtifactsSelection(selected)
+        && selected.ids.length === item.ids.length
+        && selected.ids.every(id => item.ids.includes(id))
```

Comment:
nit: probably not worth the length checks here. 
```suggestion
      return isArtifactsSelection(selected) && selected.ids.every(id => item.ids.includes(id))
```

Reply:
Yeah.. Even if you have a cluster selected, I think scaling the graph will clear the selection/popover